packages/slice-machine/lib/builders/SliceBuilder/index.tsx

```diff
@@ -74,38 +106,42 @@ const SliceBuilder: React.FunctionComponent = () => {
     }
   }, [data]);
 
-  useEffect(() => {
-    if (!store) return;
-
-    return () => store.reset();
-  }, []);
-
   const sliceView = useMemo(
-    () => [{ sliceID: Model.model.id, variationID: variation.id }],
-    [Model.model.id, variation.id]
+    () =>
+      Model && variation
+        ? [{ sliceID: Model.model.id, variationID: variation.id }]
+        : null,
+    [Model?.model.id, variation?.id]
   );
 
+  if (!Model || !variation || !sliceView) return null;
```

Comment:
if it's above it means that the function could return early and the hooks underneath would be conditionally called or not called. This is against the hooks rules since in React the hooks must always be called the same number of times and in the same order https://reactjs.org/docs/hooks-rules.html, https://bobbyhadz.com/blog/react-hook-usestate-called-conditionally#:~:text=The%20error%20%22React%20hook%20'useState,of%20how%20the%20error%20occurs.